### PR TITLE
Intern-3

### PR DESCRIPTION
**Pull request by olegstasiw**
- Create Countries List Screen
- Create a View Models
- Setup Dependency Injection to pass all needed data/services to the view model
- import Swinject
- Implement Assembly

### CODE REVIEW

File: Countries Info/ViewControllers/CountriesListViewController.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        countriesCollectionView.delegate = self
+        countriesCollectionView.dataSource = self
+    }
+    
+}
+
+extension CountriesListViewController: UICollectionViewDelegate, UICollectionViewDataSource {
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        3
```
Comment (romanmiller) [inline Countries Info/ViewControllers/CountriesListViewController.swift]: return?

File: Countries Info.xcodeproj/xcuserdata/ostasi.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 2]
```diff
@@ -91,21 +91,42 @@
 			<key>isShown</key>
 			<false/>
 			<key>orderHint</key>
-			<integer>9</integer>
+			<integer>8</integer>
 		</dict>
-		<key>Spectre (Playground) 1.xcscheme</key>
+		<key>Sample-iOS (Playground) 1.xcscheme</key>
```
Comment (romanmiller) [inline Countries Info.xcodeproj/xcuserdata/ostasi.xcuserdatad/xcschemes/xcschememanagement.plist]: Do we actually need these changes with the playground, please check.

**Review by romanmiller — APPROVED** [review]
LGTM

File: Countries Info/Asseblies/ViewControllerAssembly.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+import UIKit
+import Swinject
+
+class ViewControllerAssembly: Assembly {
+    func assemble(container: Container) {
+        container.register(CountriesListViewController.self) { r in
+            let storyboard = UIStoryboard(name: "Main", bundle: nil)
+            let controller = storyboard.instantiateViewController(withIdentifier: "CountriesListViewController") as! CountriesListViewController
+            controller.viewModel = CountriesListViewModel(apollo: DependecyInjectionManager.shared.assembler.resolver.resolve(ApolloManager.self)!)
```
Comment (trudakevych) [inline Countries Info/Asseblies/ViewControllerAssembly.swift]: The resolver is actually passed as a parameter here, so we can simplify in this way: 
`CountriesListViewModel(apollo: r.resolve(ApolloManager.self)!)`

File: Countries Info/Supporting files/AppDelegate.swift [hunk 4]
```diff
@@ -6,31 +6,22 @@
...
 import UIKit
+import Swinject
 
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
 
-
-
+    var window: UIWindow?
+    private let DIAssabler = DependecyInjectionManager.shared.assembler
```
Comment (trudakevych) [inline Countries Info/Supporting files/AppDelegate.swift]: By naming convention we need to start var name from lower character, plus there is `m` letter missing, we can shorten this war to be for example just `assembler`

File: Countries Info/Asseblies/ViewControllerAssembly.swift (hunk 3 above)
Comment (romanmiller) [reply Countries Info/Asseblies/ViewControllerAssembly.swift]: А точно ми про то говорили я щось завтикав :)

**Review by trudakevych — APPROVED** [review]

**Pull request merged**